### DISCUSSION TITLE
React Navigation v5 async storage trigger problem

### DISCUSSION THREAD

**Discussion by YunusEmreNalbant** [react-navigation/react-navigation#9446] [Q&A] [answered]
Hi,
`
```
isSignedIn ? (
  <>
    <Stack.Screen name="Home" component={HomeScreen} />
    <Stack.Screen name="Profile" component={ProfileScreen} />
    <Stack.Screen name="Settings" component={SettingsScreen} />
  </>
) : (
  <>
    <Stack.Screen name="SignIn" component={SignInScreen} />
    <Stack.Screen name="SignUp" component={SignUpScreen} />
  </>
)
```
`

as above, there is construction of the route.I keep the issignedin value in asyncstorage. When the user logs in, issignedin: true, but it doesn't automatically redirect me to the Home screen. When you do this with Redux, it automatically redirects to the Home screen, but AsyncStorage does not.. How do I solve this?

 iOS or Android : Android                
  "@react-navigation/drawer": "^5.12.4",
   "@react-navigation/native": "^5.9.3",
   "@react-navigation/stack": "^5.14.3",
    "react-native": "0.63.4",
    "react-native-gesture-handler": "^1.10.3",
    "react-native-safe-area-context": "^3.2.0",
    "react-native-screens": "^2.18.1",
     node: v14.15.4

**Comment by raajnadar** [comment, marked as answer]
You need a global state management to trigger a re render to switch between auth & no auth state. I always used context api because it is built in react js.

If you use async storage just call a state update to switch!

**Reply by YunusEmreNalbant** [reply]
thank you.
